x-pack/libbeat/management/managerV2_test.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package management
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent-client/v7/pkg/client/mock"
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+
+	"github.com/elastic/beats/v7/libbeat/common/reload"
+)
+
+func TestManagerV2(t *testing.T) {
+
+	srv := mockSrvWithUnits([]*proto.UnitExpected{
+		{
+			Id:             "output-unit",
+			Type:           proto.UnitType_OUTPUT,
+			ConfigStateIdx: 1,
+			Config: &proto.UnitExpectedConfig{
+				Id:   "default",
+				Type: "elasticsearch",
+				Name: "elasticsearch",
+			},
+			State:    proto.State_HEALTHY,
+			LogLevel: proto.UnitLogLevel_INFO,
+		},
+		{
+			Id:             "input-unit-1",
+			Type:           proto.UnitType_INPUT,
+			ConfigStateIdx: 1,
+			Config: &proto.UnitExpectedConfig{
+				Id:   "system/metrics-system-default-system-1",
+				Type: "system/metrics",
+				Name: "system-1",
+				Streams: []*proto.Stream{
+					{
+						Id: "system/metrics-system.filesystem-default-system-1",
+						Source: requireNewStruct(t, map[string]interface{}{
+							"metricsets": []interface{}{"filesystem"},
+							"period":     "1m",
+						}),
+					},
+				},
+			},
+			State:    proto.State_HEALTHY,
+			LogLevel: proto.UnitLogLevel_INFO,
```

Comment:
If you used a lower log level for one of these units you could trivially test the log level adjustment with a call to [logp.GetLevel](GetLevel).

Reply:
Added to the test to cover this.